src/Util/MigrationRollbackBatch.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\islandora_spreadsheet_ingest\Util;
+
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\migrate\MigrateMessage;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate_tools\MigrateExecutable;
+
+/**
+ * Class responsible for rolling back a migration in batches.
+ */
+class MigrationRollbackBatch extends MigrateExecutable {
+
+  use DependencySerializationTrait {
+    __sleep as traitSleep;
+    __wakeup as traitWakeup;
+  }
+
+  /**
+   * Messenger service to display messages.
+   *
+   * @var Drupal\Core\Messenger\MessengerInterface
+   */
+  protected MessengerInterface $messenger;
+
+  public function __construct(
+    MigrationInterface $migration,
+    MessengerInterface $messenger,
+    array $options,
+  ) {
+    parent::__construct($migration, new MigrateMessage(), $options);
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * Prepare a batch array for execution for the given migration.
+   *
+   * @return array
+   *   A batch array with operations and the like.
+   *
+   * @throws \Exception
+   *   If the migration could not be enqueued successfully.
+   */
+  public function prepareBatch(): array {
+    return [
+      'title' => $this->t(
+          'Rolling back migration: @migration',
+          ['@migration' => $this->migration->id()]
+      ),
+      'operations' => [
+        [[$this, 'processBatch'], []],
+      ],
+      'finished' => [$this, 'finishBatch'],
+    ];
+  }
+
+  /**
+   * Process each batch to roll back all contained rows.
+   *
+   * @param array $context
+   *   The batch context.
+   *
+   * @return void
+   *   void
+   */
+  public function processBatch(array &$context): void {
+    $context['message'] = $this->t(
+        'Processing of "@migration_id"', ['@migration_id' => $this->migration->id()]);
+
+    $status = $this->rollback();
```

Comment:
So, while this strictly does "do the rollback in a batch", this single call here could easily timeout if there's a large enough set being rolled back, which is entirely what batch structuring is supposed to prevent.

Really, what I'm thinking we need is to pull apart [the underlying `::rollback()` code](https://git.drupalcode.org/project/drupal/-/blob/10.2.x/core/modules/migrate/src/MigrateExecutable.php?ref_type=heads#L312-375), similar to how [`::import()`](https://git.drupalcode.org/project/drupal/-/blob/10.2.x/core/modules/migrate/src/MigrateExecutable.php?ref_type=heads#L146-310) was pulled apart to produce our [batched import](https://github.com/discoverygarden/dgi_migrate/blob/main/src/MigrateBatchExecutable.php#L130).

Reply:
@adam-vessey please review the code changes to address your feedback
https://github.com/discoverygarden/islandora_spreadsheet_ingest/pull/108/commits/756e22aba3840e14107c1f776fb808d314ee17c1

Reply:
Might make more sense to deal with things over on #108 ?